cypress/integration/05_collection_spec.js

```diff
@@ -1,164 +1,332 @@
 /// <reference types="Cypress" />
+import { FSApi } from '../../fusion-studio-extension/src/common/api';
 
 context('Collection Operations', () => {
-  let fetchSpy;
   describe('working with tree view', () => {
+    const connection = {
+      server: Cypress.env('API_HOST'),
+      username: 'admin',
+      password: '',
+    };
     before(() => {
+      // prepare collections used in the test
+      new Cypress.Promise(async resolve => {
+        await FSApi.remove(connection, '/db/test', true).catch(e => { });
+        await FSApi.newCollection(connection, '/db/test');
+        await FSApi.newCollection(connection, '/db/test/col1');
+        await FSApi.save(connection, '/db/test/col1/test.txt', 'test text file');
+        resolve();
+      })
       cy.connect()
       cy.visit('/');
     })
     beforeEach(() => {
-      cy.window().then(win => fetchSpy = cy.spy(win, 'fetch').as('fetch'));
+      cy.window().then(win => cy.spy(win, 'fetch').as('fetch'));
+    })
+    after(() => {
+      // delete the test colelction
+      new Cypress.Promise(resolve => FSApi.remove(connection, '/db/test', true).then(resolve).catch(resolve))
     })
 
-    describe('db context menu', () => {
-      it('should display creation options', () => {
-        cy.get('.fusion-view')
-          .should('be.visible')
-        cy.get('.fusion-item')
-          .click()
-        //  all we need is the final part of the node-id attribute
-        // (DP): start workaround for #413
-        cy.get('[node-id$=db]')
-          .click()
-        cy.get('.fa-spinner')
-          .should('not.exist')
-        // (DP): end workaround for #413
-        cy.get('@fetch').should('be.calledWith', Cypress.env('API_HOST') + '/exist/restxq/fusiondb/explorer?uri=/db');
-        cy.get('[node-id$=db]')
-          .rightclick();
-        cy.get('.p-Menu')
-          .should('be.visible')
-          .find('[data-command="fusion.new-collection"]')
-          .should('be.visible')
-          .contains('New collection')
-          .click()
-        cy.focused()
-          .type('{enter}')
-        cy.get('.fusion-view')
-          .contains('untitled-1')
-        cy.get('@fetch').should('be.calledWithMatch', Cypress.env('API_HOST') + '/exist/restxq/fusiondb/collection?uri=/db/untitled-1', { method: 'PUT' });
-      })
+    it('should display creation options', () => {
+      cy.get('.fusion-view')
+        .should('be.visible')
+      cy.get('.fusion-item')
+        .click()
+      //  all we need is the final part of the node-id attribute
+      // (DP): start workaround for #413
+      cy.get('[node-id$=db]')
+        .click()
+        .prev().should('not.have.class', 'fa-spin')
+      // (DP): end workaround for #413
+      cy.get('[node-id$=test]')
+        .click()
+        .prev().should('not.have.class', 'fa-spin')
```

Comment:
why are we testing spin here, multiple times ?

Reply:
This here is to make sure the node is done loading
I added the wait because it turned out that the spin disappears before the rest of the tree is rendered (the nodes are not rendered all at once), so the wait is to give React enough time to update the dom